math/libfaiss/Portfile

```diff
@@ -44,19 +46,127 @@ build.env-append    "SHAREDFLAGS=-Wl,-F. -bundle -undefined dynamic_lookup"
 build.env-append    FAISSSHAREDFLAGS=-dynamiclib
 build.env-append    BLASCFLAGS=-DFINTEGER=int
 build.env-append    "BLASLDFLAGS=-framework Accelerate"
+build.env-append    "CPPFLAGS=-I${prefix}/include/libomp"
 build.target        libfaiss.a libfaiss.dylib
 
-destroot.target
-destroot.destdir
-destroot.cmd        mkdir -p ${destroot}${prefix}/lib && \
-                    cp libfaiss.a libfaiss.dylib ${destroot}${prefix}/lib && \
-                    mkdir -p ${destroot}${prefix}/include/faiss && \
-                    cp -R ./*.h ${destroot}${prefix}/include/faiss/
-
-variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
-    depends_lib-append  port:OpenBLAS
-    build.env-delete    "BLASLDFLAGS=-framework Accelerate"
-    build.env-append    BLASLDFLAGS=${prefix}/lib/libopenblas.dylib
+if {${name} eq ${subport}} {
+    variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
+        depends_lib-append  port:OpenBLAS
+        build.env-delete    "BLASLDFLAGS=-framework Accelerate"
+        build.env-append    BLASLDFLAGS=${prefix}/lib/libopenblas.dylib
+    }
+
+    default_variants +openblas
+}
+
+# Python subports
```

Comment:
perhaps the example Portfile I suggested earlier today wasn't the best example, sorry.... What you've added here now is not very readable, which is understandable if the port actually needs all these variables and stuff.

I would suggest to do it a little different in this case: first just define the subports and then use in the ` if {[string match "py*" ${subport}]} {` block the `python` PG to make things easier. If you do that you can re-use all the variables like ${python.bin} and such from the PG. Perhaps something like this (of course it still needs the configure/build parts):
```
# create Python subports
foreach v {37 38} {
    subport py${v}-${github.project} {}
}

# Python bindings for supported Python versions
if {[string match "py*" ${subport}]} {
    PortGroup           python 1.0

    categories-append   math
    description         Python ${python.branch} bindings for ${description}
    long_description    This package provides Python ${python.branch} bindings, \
                        modules, and scripts.

    python.default_version ${python.version}

    depends_build-append \
                        port:py${python.version}-setuptools \
                        port:swig-python
    depends_lib-append  port:${name} \
                        port:py${python.version}-numpy
   
    <whatever else is needed>
}

Reply:
Thanks done. I had to move `python.default_version` into the `support` to avoid the default version `38` when `python.versions` is undefined.

Reply:
yes, good catch!